stdlib/public/core/StringIndex.swift

```diff
@@ -70,6 +70,11 @@ isn't frozen.
     If set, the position is known to be expressed in UTF-16 code units.
     (Introduced in Swift 5.7)
 
+  * b4: `_isWordAligned`
+
+    If set, the index is known to be on a Unicode word boundary.
+    (Introduced in Swift 5.7)
+
```

Comment:
How important is this bit? It feels a little off to be trying to add this to _every_ index, especially since the word view can have its own index type.